app/controllers/ips_controller.rb

```diff
@@ -6,29 +6,20 @@ def new
     @locations = available_locations
   end
 
-  # rubocop:disable Style/IfInsideElse
   def create
     @ip = Ip.new(create_params)
 
     if @ip.save
-      publish_for_performance_platform
-      publish_radius_whitelist
+      Services::HandleNewIp.new(ip: @ip).execute
```

Comment:
There is no way we can re-use this same controller and just check for the presence of params[:locations] ?
Would this complicate stuff? What is the Rails way of doing this?